File: ai-service/config/firebase.py
import os
import sys
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from dotenv import load_dotenv
load_dotenv(os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), '.env'))

import firebase_admin
from firebase_admin import credentials, firestore

_db = None

def initialize_firebase():
    global _db
    if not firebase_admin._apps:
        cred_path = os.getenv('FIREBASE_CREDENTIALS_PATH')
        if not cred_path:
            raise RuntimeError('FIREBASE_CREDENTIALS_PATH not set in .env')
        
        # Normalize path for Windows
        cred_path = os.path.normpath(cred_path)
        
        if not os.path.exists(cred_path):
            raise FileNotFoundError(f'Service account key not found at: {cred_path}')
        
        cred = credentials.Certificate(cred_path)
        firebase_admin.initialize_app(cred)
        print(f'✅ Firebase initialized in AI service')
    _db = firestore.client()

def get_db():
    if _db is None:
        raise RuntimeError('Firebase not initialized. Call initialize_firebase() first.')
    return _db

File: backend/src/config/firebase.js
const admin = require('firebase-admin');
const path = require('path');

let db;
let auth;

function initializeFirebase() {
  if (admin.apps.length === 0) {
    const credentialsPath = path.resolve(
      process.env.FIREBASE_CREDENTIALS_PATH || './src/config/serviceAccountKey.json'
    );

    const serviceAccount = require(credentialsPath);

    admin.initializeApp({
      credential: admin.credential.cert(serviceAccount),
    });

    console.log('✅ Firebase Admin SDK initialized');
  }

  db = admin.firestore();
  auth = admin.auth();
}

function getDb() {
  if (!db) throw new Error('Firebase not initialized. Call initializeFirebase() first.');
  return db;
}

function getAuth() {
  if (!auth) throw new Error('Firebase not initialized. Call initializeFirebase() first.');
  return auth;
}

module.exports = { initializeFirebase, getDb, getAuth };

